### PR TITLE
Support for PlatformIO library manager

### PR DESCRIPTION
**Pull request by sstaub**
This file addition includes the support for the PlatformIO library manager.
With some adaptations it can used for all the libraries.
After making the PR, the library can be registered. I can do this.
https://docs.platformio.org/en/latest/librarymanager/creating.html#register

### CODE REVIEW

**Comment by fpistm** [comment]
Thanks @sstaub

**Comment by rzr** [comment]
bump, I doubt this change will cause any harm ?

**Comment by sstaub** [comment]
This PR only add's an additional file used by PIO, it has nothing to do with ethernet code.

File: library.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "description": "Arduino library to support Ethernet for STM32 based board.",
+  "repository":
+  {
+    "type": "git",
+    "url": "https://github.com/stm32duino/STM32Ethernet"
+  },
+  "version": "1.0.4",
+  "frameworks": "arduino",
+  "platforms": "ststm32"
+}
```
Comment (fpistm) [inline library.json]: Please add:
```suggestion
    "build": {
        "libArchive": false
    }
}
```
This was discussed here with @ivankravets:
https://github.com/stm32duino/STM32FreeRTOS/pull/18/files#r282812468
as a `weak` attribute exists in this lib:
https://github.com/stm32duino/STM32Ethernet/blob/27b579a9a4bf7e7ece58014d9b4f829c1a444060/src/utility/ethernetif.c#L598

File: library.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "STM32Ethernet",
+  "keywords": "Ethernet",
+  "description": "Arduino library to support Ethernet for STM32 based board.",
+  "repository":
+  {
+    "type": "git",
+    "url": "https://github.com/stm32duino/STM32Ethernet"
+  },
+  "version": "1.0.4",
```
Comment (fpistm) [inline library.json]: Please also update version
```suggestion
  "version": "1.0.5",
```

**Comment by fpistm** [comment]
Thanks @sstaub

**Pull request merged**